game/platform.hh

```diff
@@ -25,12 +21,10 @@
 #endif
 
 struct Platform {
-enum platforms { windows, linux, macos, bsd, solaris, unix };
+enum class Host_OS { Performous_OS_Win32, Performous_OS_Linux, Performous_OS_macOS, Performous_OS_BSD, Performous_OS_Solaris, Performous_OS_Unix };
```

Comment:
I would suggest restoring the original OS names, possibly capitalized, but get rid of all the extra prefixes and underscores. This whole thing is already namespaced within `Platform` so it is unclear if even an `enum class` brings any extra value here. `Platform::Host_OS::Performous_OS_macOS` WTF?!

Reply:
I'd suggest making it enum class OS, and indeed capitalized OS names.

I'd make it an enum class to prevent warnings to happen and possible mistakes to be made. See https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum for an explanation

Reply:
@Tronic there were some macro clashes because "linux" was being expanded to 1 in other places.